cate/util/opmetainf.py

```diff
@@ -494,5 +524,35 @@ def _get_input_names(cls, inputs: Dict[str, Props]) -> List[str]:
         return input_names
 
 
-_SPHINX_PARAM_DIRECTIVE_PATTERN = re.compile(":param (?P<name>[^:]+): (?P<desc>[^:]+)")
-_SPHINX_RETURN_DIRECTIVE_PATTERN = re.compile(":returns?: (?P<desc>[^:]+)")
+def is_instance_of(value, data_type) -> bool:
+    """
+    Tests whether the given *value* is an instance of the given *data_type*.
+    The latter may be a type from the Python `typing` module.
+
+    :param value: The value to test
+    :param data_type: The data type to test against
+    :return: True, if so.
+    """
+    typing_name = repr(data_type)
+    if typing_name.startswith(_TYPING_PREFIX):
```

Comment:
I preferred if you'd invert the condition, so we could have the following lines one indent level below.